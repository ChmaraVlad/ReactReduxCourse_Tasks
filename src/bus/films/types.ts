// Core
import { CaseReducer, PayloadAction } from '@reduxjs/toolkit';

// State
export type Film = {
    title: string,
    episode_id: number,
    opening_crawl: string,
    director: string,
    producer: string,
    release_date: string,
    characters: string[],
    planets: string[],
    starships:string[],
    vehicles: string[],
    species: string[],
    created: string,
    edited: string,
    url: string
}
export type Films = Array<Film>
export type FilmsState = Films | null

// Contracts
export type BaseContact<T = any> = CaseReducer<FilmsState, PayloadAction<T>>
