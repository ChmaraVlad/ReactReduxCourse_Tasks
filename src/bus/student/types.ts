export type Student = {
    firstName: string,
    lastName: string,
    email: string,
    age: string,
    speciality: string,
    sex: string,
}

export type StudentState = {
    results: Student[];
}

export const CREATE_STUDENT = '[STUDENT] - CREATE_STUDENT';
export type createStudentAction = {
    type: typeof CREATE_STUDENT;
    payload: Student
};

export type StudentActionTypes = createStudentAction;
