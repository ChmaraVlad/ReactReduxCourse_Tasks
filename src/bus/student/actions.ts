// Types
import * as Types  from './types';

export const studentActions = {
    createStudentAction: (data: Types.Student) => {
        return {
            type:    Types.CREATE_STUDENT,
            payload: data,
        };
    },
};
