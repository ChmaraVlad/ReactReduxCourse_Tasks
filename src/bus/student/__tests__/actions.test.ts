// Actions
import { studentActions } from '../actions';

// Types
import { Student } from '../types';

const testStudent: Student = {
    firstName:  'string',
    lastName:   'string',
    email:      'string',
    age:        'string',
    speciality: 'string',
    sex:        'string',
};

describe('student actions', ()=>{
    test('should create student', ()=>{
        expect(studentActions.createStudentAction(testStudent)).toMatchSnapshot();
    });

    // test('should set fetching error', ()=>{
    //     expect(filmsActions.setFetchingError('We have an error !')).toMatchSnapshot();
    // });
});
