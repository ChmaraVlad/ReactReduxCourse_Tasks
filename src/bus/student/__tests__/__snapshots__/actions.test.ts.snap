// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`student actions should create student 1`] = `
Object {
  "payload": Object {
    "age": "string",
    "email": "string",
    "firstName": "string",
    "lastName": "string",
    "sex": "string",
    "speciality": "string",
  },
  "type": "[STUDENT] - CREATE_STUDENT",
}
`;
